src/main/java/io/aiven/kafka/connect/http/sender/request/OAuth2Form.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2021 Aiven Oy and http-connector-for-apache-kafka project contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.http.sender.request;
+
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.StringJoiner;
+
+public class OAuth2Form {
+
+    private static final String SCOPE = "scope";
+    private final StringJoiner stringJoiner = new StringJoiner("&");
+
+    private final String grantTypeFormField;
+    private final String grantType;
+
+    private final String scope;
+    private final String clientIdFormField;
+    private final String clientId;
+
+    private final String clientSecretFormField;
+    private final String clientSecret;
+
+    private OAuth2Form(
+        final String grantTypeFormField,
```

Comment:
```suggestion
        final String grantTypeProperty,
```
So we avoid having `KEY`, `FormField` and `Property` names for the same thing, does it make sense?